src/mapping/impl/BasisFunctions.hpp

```diff
@@ -1,6 +1,18 @@
 #pragma once
 
-#if defined(__NVCC__)
+#if defined(__NVCC__) || (defined(__CUDA__) && defined(__CUDA_ARCH__))
```

Comment:
```suggestion
#if defined(__CUDACC__)
```
I think this should be CUDACC, not sure if the CUDA here is required at all.

Could also be that we are missing here an apple flag. See the Ginkgo version of it

https://github.com/ginkgo-project/ginkgo/blob/75b134aa73fac11e0b085a52cadbecb31af47c9a/accessor/utils.hpp#L15-L32

Reply:
Rather than distinguishing the vendor, we could also distinguish their intention: one (CUDACC) is: which compiler is looking at the code, the other is: do I compile for host or device?

Reply:
I think this is actually missing the `__Clang__` define. https://releases.llvm.org/19.1.0/docs/CompileCudaWithLLVM.html#detecting-clang-vs-nvcc-from-code

Where and how is this compiled as device code?

Reply:
https://github.com/precice/precice/blob/dd0b7734433021339503ad4bace8309216791e6a/src/mapping/device/GinkgoRBFKernels.cpp#L74

The kernels are executed on the device and the basis functions are included here. 

> Both clang and nvcc define __CUDACC__ during CUDA compilation. You can detect NVCC specifically by looking for __NVCC__.

THere is no value for us in detecting `clang` specifically. As written above, `__NVCC__` is specific to nvcc and not compatible with clang. Both define `__CUDACC__` however.

Reply:
So, for CUDA we should just check with `_CUDACC_` if this is device code.